crates/holochain_zome_types/src/query.rs

```diff
@@ -108,41 +110,75 @@ pub struct LinkQuery {
 }
 
 #[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize, SerializedBytes)]
-/// An agents chain records returned from a agent_activity_query
+/// An agent's status and chain records returned from a [ `get_agent_activity` ]
+/// query.
 pub struct AgentActivity {
-    /// Valid actions on this chain. `(sequence, action_hash)`.
+    /// Valid actions on this chain matching the filters passed to the query,
+    /// as a vector of `(sequence_index, action_hash)` tuples. **Note**:
+    /// Warrants received from other sources are not reflected in this field;
+    /// it simply contains actions whose [ `RegisterAgentActivity` ] operations
+    /// are valid.
     pub valid_activity: Vec<(u32, ActionHash)>,
-    /// Rejected actions on this chain. `(sequence, action_hash)`.
+    /// Rejected actions on this chain matching the filters passed to the
+    /// query, as a vector of `(sequence_index, action_hash)` tuples.  **Note**:
+    /// Warrants received from other sources are not reflected in this field;
+    /// it simply contains actions whose [ `RegisterAgentActivity` ] operations
+    /// are invalid.
```

Comment:
Huh I see. Feels like more confusion arising from lacking a coherent concept of action validity.